### ISSUE TITLE
Hot network questions sidebar sites filtering broken

### ISSUE THREAD

**Issue by phuclv90** [soscripted/sox#358]
### Installed Version: 2.2.32DEV  Environment: Tampermonkey

I like the hot network questions feature. But in the list there are too many questions that belong to communities I have absolute zero interest. Therefore I'd like to hide them completely

There are scripts that do the filtering on meta stackexchange, but it'll be nice if the functionality is integrated into SOX

- [Filter "Hot Network Questions" by excluding sites](https://meta.stackexchange.com/q/225297/230282)
- [Don't show questions with ignored tags on Hot Network Questions](https://meta.stackexchange.com/q/271335/230282)

### Features Enabled 

    ["Appearance-addAuthorNameToInboxNotifications","Appearance-alignBadgesByClass","Appearance-answerTagsSearch","Appearance-colorAnswerer","Appearance-dragBounty","Appearance-highlightQuestions","Appearance-isQuestionHot","Appearance-localTimestamps","Appearance-markEmployees","Appearance-metaChatBlogStackExchangeButton","Appearance-metaNewQuestionAlert","Appearance-spoilerTip","Appearance-standOutDupeCloseMigrated","Appearance-tabularReviewerStats","Appearance-topAnswers","Appearance-unspoil","Appearance-addTimelineAndRevisionLinks","Comments-autoShowCommentImages","Comments-commentReplies","Comments-commentShortcuts","Comments-confirmNavigateAway","Comments-copyCommentsLink","Comments-moveBounty","Comments-showCommentScores","Comments-hiddenCommentsIndicator","Editing-addSBSBtn","Editing-editComment","Editing-editReasonTooltip","Editing-findAndReplace","Editing-kbdAndBullets","Editing-titleEditDiff","Editing-inlineEditorEverywhere","Flags-flagOutcomeTime","Flags-flagPercentages","Flags-flagPercentageBar","Sidebar-hideHireMe","Sidebar-hideChatSidebar","Sidebar-linkedToFrom","Sidebar-hotNetworkQuestionsFiltering","Chat-chatEasyAccess","Chat-replyToOwnChatMessages","Chat-renameChat","Voting-betterCSS","Voting-stickyVoteButtons","Voting-disableVoteButtons","Extras-alwaysShowImageUploadLinkBox","Extras-linkedPostsInline","Extras-parseCrossSiteLinks","Extras-quickAuthorInfo","Extras-shareLinksMarkdown","Extras-sortByBountyAmount","Extras-warnNotLoggedIn","Extras-hideCertainQuestions","Extras-showMetaReviewCount","Extras-copyCode","Extras-dailyReviewBar","Extras-openLinksInNewTab","Extras-showQuestionStateInSuggestedEditReviewQueue"]

**Comment by phuclv90** [comment]
I've just found out that I missed the "Filter the hot network questions by their attributes" option. But no matter how I entered the filter condition, some questions are still left out

I've tried using both page links like `https://rpg.stackexchange.com/,https://law.stackexchange.com/,https://skeptics.stackexchange.com/,https://scifi.stackexchange.com/` and page names like "Role-playing Games,Law,Science Fiction & Fantasy,Skeptics" in *Sites to block* and *Titles to hide* but none of them work

It seems the best result I got is by using a match pattern like this `https://rpg.stackexchange.com/*,https://law.stackexchange.com/*,https://skeptics.stackexchange.com/*,https://scifi.stackexchange.com/*` but even then a few questions among those sites are still shown

What's the correct way to filter out those?

**Comment by shu8** [comment]
@phuclv90 I've designed the match patterns to work like [Chrome's syntax](https://developer.chrome.com/extensions/match_patterns).

Could you try using something like `*://rpg.stackexchange.com/*`? :)

**Comment by phuclv90** [comment]
Unfortunately `*://rpg.stackexchange.com/*` gives the red error that the filter module can't be loaded

If I put only one or two patterns it seems to work correctly. But as soon as I put more patterns it won't work anymore

**Comment by phuclv90** [comment]
This is the log I got with `*://rpg.stackexchange.com/*`

```
SOX:  There was an error loading the feature "Sidebar-hotNetworkQuestionsFiltering". Please raise an issue on GitHub, and copy the following error log:
SyntaxError: nothing to repeat userscript.html:62:13
tms_d56f12a9_bfef_4326_8ba6_60d0e5d74383/</sox.error
moz-extension://f779b2f8-e0ca-48fa-9437-d563452d97d0/userscript.html:62:13
init
moz-extension://f779b2f8-e0ca-48fa-9437-d563452d97d0/userscript.html:3454:25
tms_d56f12a9_bfef_4326_8ba6_60d0e5d74383/</sox.ready/<
moz-extension://f779b2f8-e0ca-48fa-9437-d563452d97d0/userscript.html:101:33
l
moz-extension://f779b2f8-e0ca-48fa-9437-d563452d97d0/userscript.html:7:29373
a/</c<
moz-extension://f779b2f8-e0ca-48fa-9437-d563452d97d0/userscript.html:7:29677
g</< eval:2:479 ja/c[d] eval line 1 > Function:52:483 
```

I tried testing with pages that have a lot of hot questions so that it's easy to see the result. One of the examples is `https://writing.stackexchange.com/*,https://skeptics.stackexchange.com/*,https://math.stackexchange.com/*`

When I put only 2 patterns it works but when I add `https://math.stackexchange.com/*` many math questions still appear

**Comment by shu8** [comment]
@phuclv90 thanks for pasting that, I'll try fixing it ASAP!

**Comment by shu8** [comment]
@phuclv90 This should hopefully be fixed in [dev 2.2.35](https://github.com/soscripted/sox/raw/dev/sox.user.js)!

Please could you confirm if it works for you? :)

I've tried with input like:

    *://writing.stackexchange.com/*,*://skeptics.stackexchange.com/*,*://math.stackexchange.com/*

**Comment by phuclv90** [comment]
Seems it's working now. Thanks

Btw why is there a "save settings" button in the *Filter the hot network questions by their attributes* section? Shouldn't the main "Save Changes" button is enough?

**Comment by shu8** [comment]
I think it was separate originally just for the convenience (dev side!) but
making it more user friendly is on my to-do list because it's quite
confusing atm!

**Comment by stibinator** [comment]
It's not working for me FF / Windows 10. My site filter list is `*://boardgames.stackexchange.com/*,*://anime.stackexchange.com/*,*://movies.stackexchange.com/*,*://scifi.stackexchange.com/*,*://worldbuilding.stackexchange.com/*,*://rpg.stackexchange.com/*` 

Originally I was using `*rpg*,*boardgames*,…` but I was getting this error:

`SOX: There was an error loading the feature "Sidebar-hotNetworkQuestionsFiltering". Please raise an issue on GitHub, and copy the following error log:
TypeError: matchHost is undefined userscript.html:62:15
    error moz-extension://d1a3e6c7-942a-44a3-8297-87b6d46a2eee/userscript.html?id=b2e64076-4d2d-4fd7-b983-43e993a3769e:62
    init moz-extension://d1a3e6c7-942a-44a3-8297-87b6d46a2eee/userscript.html?id=b2e64076-4d2d-4fd7-b983-43e993a3769e:3883
    ready moz-extension://d1a3e6c7-942a-44a3-8297-87b6d46a2eee/userscript.html?id=b2e64076-4d2d-4fd7-b983-43e993a3769e:101
    l moz-extension://d1a3e6c7-942a-44a3-8297-87b6d46a2eee/userscript.html?id=b2e64076-4d2d-4fd7-b983-43e993a3769e:7
    c moz-extension://d1a3e6c7-942a-44a3-8297-87b6d46a2eee/userscript.html?id=b2e64076-4d2d-4fd7-b983-43e993a3769e:7
    g eval:3
    d eval line 1 > Function:55`

Once I formatted it like `*://<site>.stackexchange.com/*,…` it wasn't giving any errors in the console, but it isn't doing anything, I'm still seeing all the sites.

**Comment by shu8** [comment]
@stibinator thanks for pointing this out! It should be fixed in [dev 2.5.36](https://github.com/soscripted/sox/raw/dev/sox.user.js) now :) (the CSS class was accidentally removed in a previous commit!)
